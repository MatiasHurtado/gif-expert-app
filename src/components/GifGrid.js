// import {} from "react"
import React from "react";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
// import { getGifs } from "../helpers/getGift"

export default function GifGrid({ category }) {
  //Cuando le ponemos : es por que le estamos cambiando el valor al arreglo
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p>Loading....</p>}
      <div className ="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
}


GifGrid.propTypes ={
  category: PropTypes.string.isRequired
}