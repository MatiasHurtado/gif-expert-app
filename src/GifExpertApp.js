import React, { Fragment, useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";
const GifExpertApp = ({defaultCategories=[]}) => {
  // const categories=['One Push','Samurai X','Dragon Ball'];
  const [categories, setCategories] = useState(defaultCategories);



  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
    
      <ol>
        {   
          categories.map(category => (
               <GifGrid  key={category}category={category}/>
                
         ))
        }
      </ol>
    </Fragment>
  );
};

export default GifExpertApp;
