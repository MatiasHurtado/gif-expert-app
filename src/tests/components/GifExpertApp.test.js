import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import GifExpertApp from '../../GifExpertApp';
import GifGrid from '../../components/GifGrid';


describe('Test GiftExperApp', () => {

   

   test('Verificar Si todos los datos de GifExpertApp se renderizen completamente', () => {

      const wrapper = shallow(<GifExpertApp />)

            expect(wrapper).toMatchSnapshot();
      
   })


   test('Debe de mostrar una Lista de Categorias ', () => {
      
      const categories = ['One Punch', 'Dragon Ball'];

      const wrapper = shallow(<GifExpertApp  defaultCategories={ categories }/>)

      expect(wrapper).toMatchSnapshot()

      expect(wrapper.find('GifGrid').length).toBe( categories.length );

   })
   
   
   
})
