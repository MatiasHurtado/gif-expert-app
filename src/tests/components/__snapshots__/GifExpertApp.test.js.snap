// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GiftExperApp Debe de mostrar una Lista de Categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test GiftExperApp Verificar Si todos los datos de GifExpertApp se renderizen completamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
